From dea63d6692d512adede7c3998b9ee628ef0b09c9 Mon Sep 17 00:00:00 2001
From: jayant-yadav <jayantyadav202@gmail.com>
Date: Fri, 17 Mar 2017 22:26:43 +0530
Subject: [PATCH] after inves change

---
 mig-api/investigator_endpoints.go | 45 +++++++++++++++++++++++++--------------
 1 file changed, 29 insertions(+), 16 deletions(-)

diff --git a/mig-api/investigator_endpoints.go b/mig-api/investigator_endpoints.go
index 7723360..eb28caf 100644
--- a/mig-api/investigator_endpoints.go
+++ b/mig-api/investigator_endpoints.go
@@ -170,24 +170,37 @@ func updateInvestigator(respWriter http.ResponseWriter, request *http.Request) {
 	invperm := request.FormValue("permissions")
 	if inv.Status == "" && invperm == "" {
 		panic("No updates to the investigator were specified")
-	}
-	if inv.Status != "" {
-		// update the investigator status in database
-		err = ctx.DB.UpdateInvestigatorStatus(inv)
-		if err != nil {
-			panic(err)
-		}
-		ctx.Channels.Log <- mig.Log{OpID: opid, Desc: fmt.Sprintf("Investigator %.0f status changed to %s", inv.ID, inv.Status)}
 	} else {
-		err = json.Unmarshal([]byte(invperm), &inv.Permissions)
-		if err != nil {
-			panic(err)
-		}
-		err = ctx.DB.UpdateInvestigatorPerms(inv)
-		if err != nil {
-			panic(err)
+		if inv.Status != "" && invperm == "" {
+			// update the investigator status in database
+			err = ctx.DB.UpdateInvestigatorStatus(inv)
+			if err != nil {
+				panic(err)
+			}
+			ctx.Channels.Log <- mig.Log{OpID: opid, Desc: fmt.Sprintf("Investigator %.0f status changed to %s", inv.ID, inv.Status)}
+		} else if inv.Status == "" && invperm != "" {
+			err = json.Unmarshal([]byte(invperm), &inv.Permissions)
+			if err != nil {
+				panic(err)
+			}
+			err = ctx.DB.UpdateInvestigatorPerms(inv)
+			if err != nil {
+				panic(err)
+			}
+			ctx.Channels.Log <- mig.Log{OpID: opid, Desc: fmt.Sprintf("Investigator %.0f permissions changed", inv.ID)}
+		} else {
+			// when both status and permissions are changed
+			var errStatus = ctx.DB.UpdateInvestigatorStatus(inv)
+			var errInvperm = json.Unmarshal([]byte(invperm), &inv.Permissions)
+			if errStatus != nil || errInvperm != nil {
+				panic(err)
+			}
+			errInvperm = ctx.DB.UpdateInvestigatorPerms(inv)
+			if errInvperm != nil {
+				panic(err)
+			}
+			ctx.Channels.Log <- mig.Log{OpID: opid, Desc: fmt.Sprintf("Investigator %.0f permissions changed and status changed to %s", inv.ID, inv.Status)}
 		}
-		ctx.Channels.Log <- mig.Log{OpID: opid, Desc: fmt.Sprintf("Investigator %.0f permissions changed", inv.ID)}
 	}
 	err = resource.AddItem(cljs.Item{
 		Href: fmt.Sprintf("%s/investigator?investigatorid=%.0f", ctx.Server.BaseURL, inv.ID),
-- 
2.7.4

